### ISSUE TITLE
[Feature]: Complete transition from `role` to `cnpg.io/instanceRole` label

### ISSUE THREAD

**Issue by mnencia** [cloudnative-pg/cloudnative-pg#3943]
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

The pull request #2915 introduced a new namespaced label `cnpg.io/instanceRole` to replace the existing `role` label. However, there are still places in the code where the old label is authoritative is present.

### Describe the solution you'd like

Complete the transition by ensuring the old label is never used in the code if the new one is present.

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by armru** [comment]
we are still using the old labels in some services:
```
// CreateClusterReadOnlyService create a service insisting on all the ready pods
func CreateClusterReadOnlyService(cluster apiv1.Cluster) *corev1.Service {
	return &corev1.Service{
		ObjectMeta: metav1.ObjectMeta{
			Name:      cluster.GetServiceReadOnlyName(),
			Namespace: cluster.Namespace,
		},
		Spec: corev1.ServiceSpec{
			Type:  corev1.ServiceTypeClusterIP,
			Ports: buildInstanceServicePorts(),
			Selector: map[string]string{
				utils.ClusterLabelName: cluster.Name,
				// TODO: eventually migrate to the new label
				utils.ClusterRoleLabelName: ClusterRoleLabelReplica,
			},
		},
	}
}

// CreateClusterReadWriteService create a service insisting on the primary pod
func CreateClusterReadWriteService(cluster apiv1.Cluster) *corev1.Service {
	return &corev1.Service{
		ObjectMeta: metav1.ObjectMeta{
			Name:      cluster.GetServiceReadWriteName(),
			Namespace: cluster.Namespace,
		},
		Spec: corev1.ServiceSpec{
			Type:  corev1.ServiceTypeClusterIP,
			Ports: buildInstanceServicePorts(),
			Selector: map[string]string{
				utils.ClusterLabelName:     cluster.Name,
				utils.ClusterRoleLabelName: ClusterRoleLabelPrimary,
			},
		},
	}
}
```

I think we should first migrate those before deleting the old label

**Comment by mnencia** [comment]
In this issue, I do not propose removing the old label. The goal is to get to a state where the label exists, but nothing uses it.

**Comment by mnencia** [comment]
I've checked, and this is quite a duplicate of #4894. It has been closed by #4897
